### PR TITLE
#24 Create a Sign In Page

### PR DESCRIPTION
**Pull request by ynynl**
Enabled inputs for password and username. 
Added links for forget password and apply for driver.
OnSubmit will send a post request
Added minimum styles.

### CODE REVIEW

File: web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import styles from './SignInPage.module.scss'
+
+interface FormData {
+  username: string;
+  password: string;
+}
+
+const SignInPage: NextPage = () => {
+
+  const { handleSubmit, register } = useForm<FormData>();
```
Comment (l2silver) [inline web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx:17]: So I think what you've done here is perfect :)

For more complicated forms though, I'd like you to use https://github.com/react-hook-form/resolvers#yup . I'm going to add this to the readme. Just thought I'd make a note

File: web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+// Modules
+import { NextPage } from 'next/types';
+import { useRouter } from "next/router";
+import { useForm } from 'react-hook-form';
+// MUI Core
+import { Typography, Link, TextField, Grid, Container, Button } from '@material-ui/core';
+import styles from './SignInPage.module.scss'
+
+interface FormData {
```
Comment (l2silver) [inline web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx:10]: great!

File: web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  //handle submit 
+  const onSubmit = handleSubmit(async (data) => {
+    try {
+      const response = await fetch("/userapi", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(data)
+      });
+
+      if (response.ok) {
```
Comment (l2silver) [inline web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx:29]: haha, great

File: web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (ynynl) [reply web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx]: Thank you for pointing this out! I have changed it

File: web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx (hunk 1 above)
Comment (ynynl) [reply web-client/src/shared/components/sigin-in-page/SignInPage/SignInPage/index.tsx]: Thanks! I will look that up.

File: web-client/src/pages/signin.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { NextPage } from 'next';
+import Head from 'next/head';
+import React from 'react';
+import SignInPage from "../shared/components/SignInPage";
```
Comment (l2silver) [inline web-client/src/pages/signin.tsx:4]: Am I crazy, or is the SignInPage missing from this pull request now?

**Review by l2silver — CHANGES_REQUESTED** [review]

File: web-client/src/pages/signin.tsx (hunk 5 above)
Comment (ynynl) [reply web-client/src/pages/signin.tsx:4]: I have no idea why this happens. let me try to do another push.

Comment (ynynl) [reply web-client/src/pages/signin.tsx:4]: Okay the file is back now.

File: web-client/src/shared/components/SignInPage/SignInPage.module.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
+.header {
```
Comment (l2silver) [inline web-client/src/shared/components/SignInPage/SignInPage.module.scss:1]: last thing (hopefully). Do you mind changing the name of this file to index.module.scss

Thanks.

File: web-client/src/shared/components/SignInPage/index.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      <form onSubmit={onSubmit}>
+        <Grid container spacing={3}>
+          <Grid item xs>
+            <Grid container spacing={1}>
+              <Grid item xs={12}>
+                <Typography >
+                  Sign in
+                </Typography>
+                <TextField
+                  fullWidth
```
Comment (l2silver) [inline web-client/src/shared/components/SignInPage/index.tsx:64]: I think if you have a label, you also need to include an id prop. Do you think you could add that?

File: web-client/src/shared/components/SignInPage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <Typography>
+                  Password
+                </Typography>
+                <TextField
+                  fullWidth
+                  inputRef={register({ required: true })}
+                  label="Password"
```
Comment (l2silver) [inline web-client/src/shared/components/SignInPage/index.tsx:78]: Likewise for the id here aswell

**Review by l2silver — CHANGES_REQUESTED** [review]
Almost there, just a couple small things.

File: web-client/src/shared/components/SignInPage/index.tsx (hunk 8 above)
Comment (ynynl) [reply web-client/src/shared/components/SignInPage/index.tsx:78]: like 'id =passeword' in the textfield tag?

File: web-client/src/shared/components/SignInPage/SignInPage.module.scss (hunk 6 above)
Comment (ynynl) [reply web-client/src/shared/components/SignInPage/SignInPage.module.scss:1]: np!

File: web-client/src/shared/components/SignInPage/index.tsx (hunk 8 above)
Comment (l2silver) [reply web-client/src/shared/components/SignInPage/index.tsx:78]: yep

File: web-client/src/shared/components/SignInPage/index.tsx (hunk 7 above)
Comment (l2silver) [reply web-client/src/shared/components/SignInPage/index.tsx:64]: https://www.w3schools.com/tags/att_label_for.asp . This is important for accessibility purposes, as people with screen readers will have more information about the input

Comment (ynynl) [reply web-client/src/shared/components/SignInPage/index.tsx:64]: Thank you! I have made the changes you suggested.

**Review by l2silver — APPROVED** [review]

**Pull request merged**